src/main/java/dev/waterdog/waterdogpe/network/downstream/ConnectedDownstreamHandler.java

```diff
@@ -45,6 +46,22 @@ public final boolean handle(RemoveObjectivePacket packet) {
         return false;
     }
 
+    @Override
+    public final boolean handle(SetScorePacket packet) {
+        for(ScoreInfo info : packet.getInfos()) {
+            this.player.getScoreInfos().stream().
+                    filter(info1 -> info1.getScoreboardId() == info.getScoreboardId()).
+                    findFirst().ifPresent(info1 -> this.player.getScoreInfos().remove(info1));
+        }
+
+        if(packet.getAction() == SetScorePacket.Action.SET) {
+            for(ScoreInfo info : packet.getInfos()) {
+                this.player.getScoreInfos().add(info);
+            }
+        }
```

Comment:
If old entries existed with the same scoreboard id they got removed so, we need to add the new ones